--- os-prober-1.76/linux-boot-probes/mounted/common/40grub2~	2017-09-29 23:50:52.415348657 +0100
+++ os-prober-1.76/linux-boot-probes/mounted/common/40grub2	2017-12-08 16:50:12.082601278 +0000
@@ -78,7 +78,7 @@
 				fi
 			;;
 			initrd*)
-				initrd="$(echo "$2" | sed 's/(.*)//')"
+				initrd="$(echo "$2 $3" | sed 's/(.*)//')"
 				# Initrd same.
 				if [ "$partition" != "$bootpart" ]; then
 					initrd="/boot$initrd"
