diff --git a/common.sh b/common.sh
index ffe7f47..a1721e4 100644
--- a/common.sh
+++ b/common.sh
@@ -151,7 +151,7 @@ parse_proc_mounts () {
 		set -f
 		set -- $line
 		set +f
-		printf '%s %s %s\n' "$(mapdevfs "$1")" "$2" "$3"
+		printf '%s %s %s %s\n' "$(mapdevfs "$1")" "$2" "$3" "$1"
 	done
 }
 
diff --git a/linux-boot-prober b/linux-boot-prober
index 15c2ba4..c1998c6 100755
--- a/linux-boot-prober
+++ b/linux-boot-prober
@@ -167,7 +167,7 @@ else
 			bootpart="${mountboot%% *}"
 			bootmounted="${mountboot#* }"
 		else
-			bootpart="$partition"
+			bootpart="$(grep " $mpoint/boot " "$OS_PROBER_TMP/mounted-map" | head -n1 | cut -d " " -f 4)"
 			bootmounted=0
 		fi
 		for test in /usr/libexec/linux-boot-probes/mounted/*; do
