diff -Naur os-prober-1.56/os-prober os-prober-new/os-prober
--- os-prober-1.56/os-prober	2012-09-17 11:08:12.000000000 -0400
+++ os-prober-new/os-prober	2012-10-31 09:35:51.461933470 -0400
@@ -6,6 +6,11 @@
 newns "$@"
 require_tmpdir
 
+for device in /dev/sd*; do
+        mkdir -p /run/os-prober$device/
+        mount -r $device /run/os-prober$device >/dev/null 2>&1 || :
+done
+
 log_output () {
 	if type log-output >/dev/null 2>&1; then
 		log-output -t os-prober --pass-stdout $@
