diff --git a/linux-boot-probes/mounted/common/90fallback b/linux-boot-probes/mounted/common/90fallback
index 6b2c125..d3b9730 100755
--- a/linux-boot-probes/mounted/common/90fallback
+++ b/linux-boot-probes/mounted/common/90fallback
@@ -33,7 +33,7 @@ for kernpat in /vmlinuz /vmlinux /boot/vmlinuz /boot/vmlinux "/boot/vmlinuz*" \
 			# Dracut initramfses are named differently again.
 			initrdname3=$(echo "$kernfile" | sed "s/vmlinu[zx]/initramfs\*/" | sed 's/$/.img/')
 			# And Gentoo's also
-			initrdname4=$(echo "$kernfile" | sed "s/kernel/initramfs\*/")
+			initrdname4=$(echo "$kernfile" | sed "s/kernel\|vmlinu[zx]/initramfs\*/")
 			# Also Alpine
 			initrdname5=$(echo "$kernfile" | sed "s/vmlinu[zx]/initramfs\*/")
 			foundinitrd=0
