diff -up os-prober-1.80/os-prober.15~ os-prober-1.80/os-prober
--- os-prober-1.80/os-prober.15~	2022-06-25 17:04:23.360754026 +0200
+++ os-prober-1.80/os-prober	2022-06-25 17:04:23.494085736 +0200
@@ -170,6 +170,11 @@ for partition in $(partitions); do
 	# be handled by 50mounted-tests so we can do a subvol only once.
 	type=$(blkid -o value -s TYPE $mapped || true)
 	if [ "$type" = btrfs ]; then
+                mpoint=$(grep "^$mapped " "$OS_PROBER_TMP/mounted-map" | head -n1 | cut -d " " -f 2)
+                mpoint="$(unescape_mount "$mpoint")"
+                if [ "$mpoint" = "/target/boot" ] || [ "$mpoint" = "/target" ] || [ "$mpoint" = "/" ]; then
+                       continue
+                fi
 		uuid=$(blkid -o value -s UUID $mapped)
 		if grep -q "^$uuid" "$OS_PROBER_TMP/btrfs-vols" ; then
 			continue
diff -up os-prober-1.80/os-probes/mounted/common/90linux-distro.15~ os-prober-1.80/os-probes/mounted/common/90linux-distro
--- os-prober-1.80/os-probes/mounted/common/90linux-distro.15~	2022-06-25 17:04:23.494085736 +0200
+++ os-prober-1.80/os-probes/mounted/common/90linux-distro	2022-06-25 17:09:12.620544324 +0200
@@ -31,6 +31,12 @@ if (ls "$dir"/lib*/ld*.so* && [ -d "$dir
 		long="$(printf "Debian GNU/Linux (%s)\n" "$(cat "$dir/etc/debian_version")")"
 	# RPM derived distributions may also have a redhat-release or
 	# mandrake-release, so check their files first.
+	elif [ -e "$dir/etc/openmandriva-release" ]; then
+		short="OpenMandriva Lx"
+		long="$(cat "$dir/etc/openmandriva-release")"
+	elif [ -e "$dir/etc/mageia-release" ]; then
+		short="Mageia"
+		long="$(cat "$dir/etc/mageia-release")"
 	elif [ -e "$dir/etc/altlinux-release" ]; then
 		short="ALTLinux"
 		long="$(cat "$dir/etc/altlinux-release")"
