diff --git a/linux-boot-probes/common/50mounted-tests b/linux-boot-probes/common/50mounted-tests
index 2c8aa81..3a71782 100755
--- a/linux-boot-probes/common/50mounted-tests
+++ b/linux-boot-probes/common/50mounted-tests
@@ -48,11 +48,11 @@ if [ ! -d "$tmpmnt" ]; then
 fi
 
 mounted=
-if type grub-mount >/dev/null 2>&1 && \
-   type grub-probe >/dev/null 2>&1 && \
-   grub-mount "$partition" "$tmpmnt" 2>/dev/null; then
+if type grub2-mount >/dev/null 2>&1 && \
+   type grub2-probe >/dev/null 2>&1 && \
+   grub2-mount "$partition" "$tmpmnt" 2>/dev/null; then
 	mounted=1
-	type="$(grub-probe -d "$partition" -t fs)"
+	type="$(grub2-probe -d "$partition" -t fs)"
 	[ "$type" ] || type=fuseblk
 fi
 
diff --git a/os-probes/common/50mounted-tests b/os-probes/common/50mounted-tests
index a3a7080..605619a 100755
--- a/os-probes/common/50mounted-tests
+++ b/os-probes/common/50mounted-tests
@@ -58,11 +58,11 @@ if [ ! -d "$tmpmnt" ]; then
 fi
 
 mounted=
-if type grub-mount >/dev/null 2>&1 && \
-   type grub-probe >/dev/null 2>&1 && \
-   grub-mount "$partition" "$tmpmnt" 2>/dev/null; then
+if type grub2-mount >/dev/null 2>&1 && \
+   type grub2-probe >/dev/null 2>&1 && \
+   grub2-mount "$partition" "$tmpmnt" 2>/dev/null; then
 	mounted=1
-	type="$(grub-probe -d "$partition" -t fs)" || true
+	type="$(grub2-probe -d "$partition" -t fs)" || true
 	if [ "$type" ]; then
 		debug "mounted using GRUB $type filesystem driver"
 	else
