diff --git a/os-probes/common/50mounted-tests b/os-probes/common/50mounted-tests
index 7865b51..9d878c0 100755
--- a/os-probes/common/50mounted-tests
+++ b/os-probes/common/50mounted-tests
@@ -54,9 +54,20 @@ if which grub-mount >/dev/null 2>&1 && \
 else
 	ro_partition "$partition"
 	for type in $types $delaytypes; do
-		if mount -o ro -t "$type" "$partition" "$tmpmnt" 2>/dev/null; then
+		if [ "$type" = ufs ]; then
+			for ufstype in ufs2 44bsd; do
+				if mount -o ro,ufstype=$ufstype -t "$type" "$partition" "$tmpmnt" 2>/dev/null; then
+					mounted=1
+				fi
+			done
+		else
+			if mount -o ro -t "$type" "$partition" "$tmpmnt" 2>/dev/null; then
+				mounted=1
+			fi
+		fi
+
+		if [ "$mounted" ]; then
 			debug "mounted as $type filesystem"
-			mounted=1
 			break
 		fi
 	done
diff --git a/os-probes/mounted/common/90bsd-distro b/os-probes/mounted/common/90bsd-distro
new file mode 100755
index 0000000..b190aab
--- /dev/null
+++ b/os-probes/mounted/common/90bsd-distro
@@ -0,0 +1,45 @@
+#!/bin/sh
+# Test for *BSD distributions.
+set -e
+
+. /usr/share/os-prober/common.sh
+
+partition="$1"
+dir="$2"
+type="$3"
+
+if [ "$type" = ufs ]; then
+  disk=$(echo $partition | gawk '{ match($0, /([[:alpha:][:punct:]]+)[[:digit:]]+/, disk); print disk[1] }')
+
+  if [ ! -z "$disk" ]; then
+    tpartition=$(echo $partition | sed 's|\/|\\/|g')
+
+    system=$(fdisk -l $disk | awk '/'$tpartition'[[:blank:]]+\*[[:blank:]]+.+[[:blank:]]+.+BSD/ {print $7}')
+
+    if [ ! -z "$system" ]; then
+      title=
+
+      if [ -f $dir/etc/motd ]; then
+        case $system in
+        FreeBSD | NetBSD | OpenBSD) title=$(cat $dir/etc/motd | gawk '{ match($0, /('$system')[[:blank:]]+([[:graph:]]+)[[:blank:]]+(\([[:print:]]+\))/, title); print title[1], title[2], title[3]; exit 0}')
+                                    ;;
+        esac
+      fi
+
+      if [ -z "$title" ]; then
+        title="$system"
+      fi
+
+      label="$(count_next_label "$system")"
+      echo "$partition:$title:$label:chain"
+
+      exit 0
+    else
+      exit 1
+    fi
+  else
+    exit 1
+  fi
+else
+  exit 1
+fi
